pkgs/dart_mcp_server/lib/src/utils/sdk.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+
+/// An interface class that provides a single getter of type [Sdk].
+///
+/// This provides information about the Dart and Flutter sdks, if available.
+abstract interface class SdkSupport {
+  Sdk get sdk;
+}
+
+/// Information about the Dart and Flutter SDKs, if available.
+class Sdk {
+  /// The path to the root of the Dart SDK.
+  final String? dartSdkPath;
+
+  /// The path to the root of the Flutter SDK.
+  final String? flutterSdkPath;
+
+  Sdk({this.dartSdkPath, this.flutterSdkPath});
+
+  /// Creates an [Sdk] from the path to the Dart SDK.
+  ///
+  /// If no [dartSdkPath] is given, it will attempt to find it using
+  /// [Platform.resolvedExecutable], and assuming that is the `dart` binary
+  /// under the `bin` dir of the Dart SDK.
+  ///
+  /// Validates that the path is valid by checking for the `version` file.
+  ///
+  /// If no [flutterSdkPath] is given, it will search up to see if it is nested
+  /// inside a Flutter SDK.
+  factory Sdk.find({String? dartSdkPath, String? flutterSdkPath}) {
+    // Assume that we are running from the Dart SDK bin dir if not given any
+    // other configuration.
+    dartSdkPath ??= p.dirname(p.dirname(Platform.resolvedExecutable));
+
+    final versionFile = dartSdkPath.child('version');
+    if (!File(versionFile).existsSync()) {
+      throw ArgumentError('Invalid Dart SDK path: $dartSdkPath');
```

Comment:
If `dartSdkPath` that was passed in as a parameter fails this check, perhaps we should fail more gracefully by trying again with `dartSdkPath = p.dirname(p.dirname(Platform.resolvedExecutable));`?

Reply:
I think if one is explicitly configured we should always use that one and not silently fall back on another one (most people won't see MCP server output).